set-up.sh

```diff
@@ -0,0 +1,106 @@
+SHOPIFY_STORE=$SHOPIFY_STORE
+THEME_ID=$THEME_ID
+SHOPIFY_ACCESS_TOKEN=$SHOPIFY_ACCESS_TOKEN
+EOF
+    echo "Environment variables have been set successfully in local.env."
+}
+
+# Function to load existing values from local.env
+load_existing_values() {
+    # Load existing values from local.env
+    source $ENV_FILE
+    echo "Existing environment variables found:"
+    echo "SHOPIFY_STORE: $SHOPIFY_STORE"
+    echo "THEME_ID: $THEME_ID"
+    echo "SHOPIFY_ACCESS_TOKEN: $SHOPIFY_ACCESS_TOKEN"
+}
+
+# Check if local.env file exists
+if [ -f "$ENV_FILE" ]; then
+    # Load and display existing values
+    load_existing_values
+
+    # Ask user if they want to use the existing values or enter new ones
+    read -p "Do you want to use the existing values? (yes to confirm, no to re-enter): " confirm
+    if [[ "$confirm" != "yes" ]]; then
+        prompt_for_values
+    else
+        echo "Using existing values from local.env."
+    fi
+else
+    # Prompt for new values if local.env does not exist
+    prompt_for_values
+fi
+
+# Check if .gitignore exists, if not, create it
+if [ ! -f ".gitignore" ]; then
+    touch .gitignore
+fi
+
+# If theme folder does not exist, create it
+if [ ! -d "theme" ]; then
+    mkdir theme
+    echo "Created 'theme' directory to store theme files."
+fi
+
+# Add local.env to .gitignore if not already there
+if ! grep -qx "local.env" .gitignore; then
+    echo "local.env" >> .gitignore
+    echo "Added 'local.env' to .gitignore to prevent it from being tracked by Git."
+else
+    echo "'local.env' is already listed in .gitignore."
+fi
+
+# Check and set environment variables for USER_ID and GROUP_ID
+# Only set them if they are not already set and readonly
+if [ -z "$USER_ID" ]; then
+    export USER_ID=$(id -u)
+    echo "USER_ID: $USER_ID"
+else
+    echo "USER_ID is already set: $USER_ID"
+fi
+
+if [ -z "$GROUP_ID" ]; then
+    export GROUP_ID=$(id -g)
+    echo "GROUP_ID: $GROUP_ID"
+else
+    echo "GROUP_ID is already set: $GROUP_ID"
+fi
+
+# Define the alias command
+ALIAS_COMMAND='alias shopify="docker exec -it shopify-cli-container shopify"'
+
+# Check if the alias already exists in the shell profile
+if ! grep -Fxq "$ALIAS_COMMAND" ~/.bashrc && ! grep -Fxq "$ALIAS_COMMAND" ~/.zshrc; then
+    # Append the alias to the appropriate shell profile
+    if [[ $SHELL == *"bash"* ]]; then
+        echo "$ALIAS_COMMAND" >> ~/.bashrc
+        source ~/.bashrc
+    elif [[ $SHELL == *"zsh"* ]]; then
+        echo "$ALIAS_COMMAND" >> ~/.zshrc
+        source ~/.zshrc
+    else
+        echo "Unsupported shell. Please manually add the alias to your shell profile."
+    fi
+    echo "Alias 'shopify' added and shell profile reloaded."
+else
+    echo "Alias 'shopify' already exists in your shell profile."
+fi
+
+# Inform the user to rebuild and restart Docker Compose services
+echo -e "\nPlease rebuild and restart Docker Compose services with the following commands:"
+echo -e "docker-compose build --no-cache"
```

Comment:
could we drop the `--no-cache`?

Reply:
We could certainly consider dropping the --no-cache option. However, my thinking is that building the Docker image locally is typically a one-time process. Using --no-cache helps ensure that if something does go wrong, any issues related to cached layers, like outdated dependencies or misconfigurations, are avoided. By forcing a fresh build, we can be more confident that all scripts and dependencies are installed and configured correctly, reducing the risk of hidden problems that might otherwise be carried over from the cache.

Reply:
Docker is able to utilize cache layers correctly if we use them correctly, using `--no-cache` feels a bit backwards because we've historically moved away from it.

Ideally we don't want users to be building this locally anyway.